batchflow/models/metrics/regression.py

```diff
@@ -0,0 +1,131 @@
+        # Since we are not interested in across batch aggregation we only need to concatenate this arrays along 0 axis
+        # In this scenario tho we can mess with 2 cases:
+        # 1. targets is a single batch of multioutput targets and has the following structure [[], [], ..].
+        # 2. targets is a list of data come from couple of batches in a single output task
+        # and has the same structure [[], [], ..].
+        # In the first case concatenating such arrays do not have any sence. To separate such cases we introduce
+        # 'multi' argument.
+
+        if np.ndim(targets) == 1 or (np.ndim(targets) == 2 and multi):
+            self.targets = np.array(targets)
+            self.predictions = np.array(predictions)
+        else:
+            self.targets = np.concatenate(targets, axis=0)
+            self.predictions = np.concatenate(predictions, axis=0)
+
+        self.weights = np.array(weights).flatten()
+        self.gap = gap
+        self.multi = multi
+
+        self._agg_fn_dict = {
+            'mean': lambda x: np.mean(x),
+        }
+
+    def __getattr__(self, name):
+        if name == "METRICS_ALIASES":
+            raise AttributeError # See https://nedbatchelder.com/blog/201010/surprising_getattr_recursion.html
+        name = METRICS_ALIASES.get(name, name)
+        return object.__getattribute__(self, name)
+
+    def mean_absolute_error(self):
+        return np.average(np.abs(self.predictions - self.targets), axis=0,
+                          weights=self.weights)
+
+    def mean_squared_error(self):
+        return np.average((self.predictions - self.targets) ** 2, axis=0,
+                          weights=self.weights)
+
+    def median_absilute_error(self):
+        return np.median(np.abs(self.predictions - self.targets))
+
+    def max_error(self):
+        return np.max(np.abs(self.predictions - self.targets))
+
+    def root_mean_square_error(self):
+        return np.sqrt(self.mean_squared_error())
+
+    def r2_score(self):
+        """ r2_score """
+        if self.weights is not None:
+            weight = self.weights[:, np.newaxis]
+        else:
+            weight = 1
+        numerator = (weight * (self.predictions - self.targets) ** 2).sum(axis=0)
+        targets_avg = np.average(self.targets, axis=0, weights=self.weights)
+        denominator = (weight * (self.targets - targets_avg) ** 2).sum(axis=0)
+        return 1 - (numerator / denominator)
+
+    def explained_variance_ratio(self):
+        """ explained_variance """
+        diff_avg = np.average(self.predictions - self.targets, axis=0, weights=self.weights)
+        numerator = np.average(self.predictions - self.targets - diff_avg, axis=0, weights=self.weights)
+        targets_avg = np.average(self.targets, axis=0, weights=self.weights)
+        denominator = np.average((self.targets - targets_avg) ** 2, axis=0, weights=self.weights)
+        return 1 - (numerator / denominator)
+
+    def accuracy(self):
```

Comment:
It might be convenient to have `gap` parameter when calculating accuracy, e.g. to compare how acc is changed with varying the gap